monitor/monitoring-dam-generator-core/src/main/java/eu/seaclouds/monitor/monitoringDamGenerator/MonitoringDamGenerator.java

```diff
@@ -0,0 +1,249 @@
+
+                tempDataCollectorDeploymentScripts = new HashMap<String, String>();
+                MODACloudsDcDeploymentScriptGenerator modacloudsDcScriptGen = new MODACloudsDcDeploymentScriptGenerator();
+                Module mainModule = null;
+
+                if (host.getDeploymentType().equals("IaaS")) {
+                    tempRules = irg.generateMonitoringRules(host
+                            .getHostName());
+
+                    host.setRules(tempRules);
+                    
+                    mainModule = getMainHostedModule(modules, host);
+                    String modacloudsDeploymentScript = modacloudsDcScriptGen
+                            .generateDataCollectorDeploymentScript(mainModule,
+                                    monitoringManagerIp, monitoringManagerPort);
+                    tempDataCollectorDeploymentScripts.put(MODACLOUDS_DC_ID,
+                            modacloudsDeploymentScript);
+                }
+
+                host.setDataCollectorBashDeploymentScripts(tempDataCollectorDeploymentScripts);
+
+            }
+
+            for (Module module : modules) {
+
+                logger.info("Generating monitoring information for module "
+                        + module.getModuleName());
+
+                tempRules = arg.generateMonitoringRules(module);
+                tempRules.getMonitoringRules().addAll(
+                        nrg.generateMonitoringRules(module)
+                                .getMonitoringRules());
+
+                module.setRules(tempRules);
+
+                tempDataCollectorDeploymentScripts = new HashMap<String, String>();
+
+                if (module.isJavaApp()) {
+
+                    JavaAppLevelDcDeploymentScriptGenerator javaDcScriptGen = new JavaAppLevelDcDeploymentScriptGenerator();
+                    String javaDcDeploymentScript = javaDcScriptGen
+                            .generateDataCollectorDeploymentScript(module,
+                                    monitoringManagerIp, monitoringManagerPort);
+                    tempDataCollectorDeploymentScripts.put(JAVA_APP_DC_ID,
+                            javaDcDeploymentScript);
+
+                }
+
+                if (CustomizedApplications.isCustomized(module.getModuleName())) {
+
+                    NuroDcDeploymentScriptGenerator nuroDcScriptGen = new NuroDcDeploymentScriptGenerator();
+                    String nuroDcDeploymentScript = nuroDcScriptGen
+                            .generateDataCollectorDeploymentScript(module,
+                                    monitoringManagerIp, monitoringManagerPort);
+                    tempDataCollectorDeploymentScripts.put(NURO_DC_ID,
+                            nuroDcDeploymentScript);
+
+                }
+
+                module.setDataCollectorBashDeploymentScripts(tempDataCollectorDeploymentScripts);
+                toReturn.add(module);
+
+            }
+
+            this.generateDamMonitoringInfo(toReturn);
```

Comment:
The difference is that generateMonitoringInfo just geneartes bash scripts that the are pointed in the interface field of the TOSCA yaml specific of a DC. The generateDamMonitoringInfo method is generating the TOSCA yaml required for that DC. Maybe I can give the method a more meaningful name or move what generateDamMonitoringInfo is doing in the toString of the Module itself.